inference-engine/thirdparty/clDNN/tests/test_cases/reduce_gpu_test.cpp

```diff
@@ -1611,3 +1616,232 @@ TEST(reduce_gpu, common_bfwzyx_log_sum_exp_keepdims) {
+        else
+            layout_format = format::bfwzyx;
+
+        using input_t = typename input_data_type<InputT>::type;
+        using output_t = typename output_data_type<OutputT>::type;
+
+        auto input_size = tensor(batch(batch_num), feature(input_f), spatial(input_x, input_y, input_z, input_w));
+        auto input_data = generate_random_6d<input_t>(batch_num, input_f, input_x, input_y, input_z, input_w, 1, 5, 9);
+        auto input_lay = layout(input_dt, layout_format, input_size);
+        auto input_mem = memory::allocate(engine, input_lay);
+
+        {
+            auto input_ptr = input_mem.pointer<input_t>();
+            for (int fi = 0; fi < input_f; fi++)
+                for (int wi = 0; wi < input_w; wi++)
+                    for (int zi = 0; zi < input_z; zi++)
+                        for (int yi = 0; yi < input_y; yi++)
+                            for (int xi = 0; xi < input_x; xi++) {
+                                for (int bi = 0; bi < batch_num; bi++) {
+                                    tensor coords = tensor(batch(bi), feature(fi), spatial(xi, yi, zi, wi));
+                                    size_t offset = input_lay.get_linear_offset(coords);
+                                    input_ptr[offset] = input_data[bi][fi][xi][yi][zi][wi];
+                                }
+                            }
+        }
+
+        std::vector<cldnn::reduce_mode> modes {
+            cldnn::reduce_mode::max,
+            cldnn::reduce_mode::min,
+            cldnn::reduce_mode::mean,
+            // reduce_mode::prod,
+            cldnn::reduce_mode::sum,
+            cldnn::reduce_mode::logical_and,
+            cldnn::reduce_mode::logical_or,
+            // reduce_mode::sum_square,
+            cldnn::reduce_mode::l1,
+            // reduce_mode::l2,
+            // reduce_mode::log_sum,
+            cldnn::reduce_mode::log_sum_exp
+        };
+
+        for (auto& target_mode : modes)
+        {
+            auto reference_result = reference_reduce(input_data, target_mode, reduce_axis, batch_num,
+                                                    input_f, input_w, input_z, input_y,
+                                                    input_x, input_dim, keep_dims);
+
+            topology topology;
+            auto red = reduce("reduce", "input", target_mode, reduce_axis, keep_dims);
+            if (force_output_dt) {
+                red.output_data_type = output_dt;
+            }
+            topology.add(input_layout("input", input_mem.get_layout()));
+            topology.add(red);
+            build_options options;
+            options.set_option(build_option::optimize_data(true));
+            implementation_desc reduce_impl = {input_format, kernel_name};
+            options.set_option(build_option::force_implementations({{"reduce", reduce_impl}}));
+            network network(engine, topology, options);
+            network.set_input_data("input", input_mem);
+
+#if REDUCE_PROF
```

Comment:
This profiling code seems to be not needed, but I think we can consider implementing of some generic performance benchmark helper for cldnn::network. Anyway, my opinion is that the profiling code must be removed from this test.

Reply:
I see.